thespian/system/multiprocCommon.py

```diff
@@ -512,6 +510,69 @@ def shutdown_signal_detected(signum, frame):
     return shutdown_signal_detected
 
 
+def get_min_log_level(logDefs):
+    """
+    Determine the minimum logging level based on the rules in:
+    https://docs.python.org/3/library/logging.config.html#logging.config.dictConfig
+
+    Note: The level and handlers entries are interpreted as for the root logger,
+    except that if a non-root logger’s level is specified as NOTSET, the system
+    consults loggers higher up the hierarchy to determine the effective
+    level of the logger.
+
+    :param logDefs: a logging configuration dictionary object
+    :return: integer value of the lowest log level, or the default level: logging.WARNING.
+
+    >>> import logging
+    >>> logDefs = {'version': 1, 'handlers': {'h1': {'class':
+    ... 'logging.FileHandler', 'filename': 'example.log', 'formatter':
+    ... 'normal', 'filters': ['notActorLog'], 'level': "INFO"}, 'h2': {'class':
+    ... 'logging.FileHandler', 'filename': 'example.log', 'formatter':
+    ... 'actor', 'filters': ['isActorLog'], 'level': "INFO"}},
+    ... 'loggers': {'': {'handlers': ['h1', 'h2'], 'level': "DEBUG"}}}
+    >>> get_min_log_level(logDefs) == logging.DEBUG
+    True
+    >>> logDefs = {'version': 1, 'loggers': {'root' : {'level': 10}, 'level': 20}}
+    >>> min1 = get_min_log_level(logDefs)
+    >>> logDefs = {'version': 1, 'loggers': {'root': {'level': 20}, 'level': 10}}
+    >>> min2 = get_min_log_level(logDefs)
+    >>> min1 == min2 == 10
+    True
+    >>> get_min_log_level({"version" : 1}) == logging.WARNING # default log level
+    True
+    >>> logDefs = {'version': 1, 'loggers': {'root': {'level': "INFO"},
+    ... 'handler2': {"level": logging.NOTSET}}}
+    >>> get_min_log_level(logDefs) == logging.INFO
+    True
```

Comment:
The `root` key is actually a top-level key in the Python logging dictionary schema, instead of appearing under `loggers` (yes, the schema is a bit awkward).  If you update this last doctest, I think it will reveal a test failure, but if you look at my comment associated with line 568, the change I'm suggesting there should resolve this failure.

Reply:
Good catch.
Wish there was a proper, official Json schema.